src/Float/Extra.elm

```diff
@@ -343,6 +349,20 @@ aboutEqual a b =
         abs (a - b) <= 1.0e-5 + 1.0e-8 * abs a
 
 
+{-| Check if two values are equal within a given tolerance.
+
+    Float.Extra.equalWithin 1.0e-6 1.9999 2.0001
+    --> False
+
+    Float.Extra.equalWithin 1.0e-3 1.9999 2.0001
+    --> True
+
+-}
+equalWithin : Float -> Float -> Float -> Bool
```

Comment:
I kind of wonder if a elm-test style `type Tolerance = Relative Float | Absolute Float | RelativeOrAbsolute Float Float` wouldn't be nicer here as the first argument?

Reply:
Perhaps...I've always been suspicious of relative tolerances myself, but that may be because I personally often work with values that are positions/coordinates instead of fundamentally being magnitudes. When working with positions it really doesn't make sense to use a relative tolerance, since a "small" value is just one that happens to be close to your chosen origin point, which is arbitrary. (Said another way, if I'm comparing the positions of two points in space to see if they're approximately equal, that comparison should _not_ depend on my choice of origin point - but if I use a relative tolerance to compare coordinate values, then the comparison _will_ depend on the choice of origin point.)

My hunch is that in most cases it's better to choose a tolerance value based on some higher-level context than individual points/values (e.g. the size of a geometric bounding box, or more generally some nominal value representing the general scale of things that you're working with), and then use that fixed tolerance as an 'absolute' tolerance for all comparisons. (If relative tolerances are so great, then why do you usually have to combine them with a 'fudge factor' for comparisons near zero?)

Of course, all that said, with an API like you propose people like me can just choose to use `Absolute` and others can decide for themselves if `Relative` or `RelativeOrAbsolute` is appropriate for their use case 🙂

Reply:
On that note...if we do use a relative tolerance, should it be based on `abs firstValue` or `max (abs firstValue) (abs secondValue)`?

- The first case (as used currently by `aboutEqual`) has the advantage that all comparisons against the same reference value can use the same tolerance
- The second case has the advantage that the comparison is symmetric, e.g. `aboutEqual a b` will always return the same result as `aboutEqual b a`, which is not true of the current implementation

Reply:
This is a great comment and an excellent explanation! I think the best action really would be to adapt it into extra documentation for `aboutEqual` as guidance on the correctness advantages of switching to `equalWithin` and how to choose a correct tolerance parameter.